cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -386,33 +392,55 @@ ignoringSeverity :: Tracer IO a -> Tracer IO (WithSeverity a)
 ignoringSeverity tr = Tracer $ \(WithSeverity _ ev) -> traceWith tr ev
 {-# INLINE ignoringSeverity #-}
 
+countMyBlocks
+  :: forall blk. ()
+  => HasHeader (Header blk)
+  => BlockSupportsMetrics blk
+  => BlockConfig blk
+  -> AF.AnchoredFragment (Header blk)
+  -> Int64
+countMyBlocks blockConfig = fromIntegral . length . AS.filter isMine
+  where isMine :: Header blk -> Bool
+        isMine header = isSelfIssued blockConfig header == IsSelfIssued
+
 traceChainMetrics
-  :: forall blk. HasHeader (Header blk)
-  => Trace IO Text -> Tracer IO (ChainDB.TraceEvent blk)
-traceChainMetrics tr = Tracer $ \ev ->
-  fromMaybe (pure ()) $
-    doTrace <$> chainTipInformation ev
- where
-   chainTipInformation :: ChainDB.TraceEvent blk -> Maybe ChainInformation
-   chainTipInformation = \case
-     ChainDB.TraceAddBlockEvent ev -> case ev of
-       ChainDB.SwitchedToAFork _warnings newTipInfo _ newChain ->
-         Just $ chainInformation newTipInfo newChain
-       ChainDB.AddedToCurrentChain _warnings newTipInfo _ newChain ->
-         Just $ chainInformation newTipInfo newChain
-       _ -> Nothing
-     _ -> Nothing
-
-   doTrace :: ChainInformation -> IO ()
-   doTrace ChainInformation { slots, blocks, density, epoch, slotInEpoch } = do
-     -- TODO this is executed each time the chain changes. How cheap is it?
-     meta <- mkLOMeta Critical Public
-
-     traceD tr meta "density"     (fromRational density)
-     traceI tr meta "slotNum"     slots
-     traceI tr meta "blockNum"    blocks
-     traceI tr meta "slotInEpoch" slotInEpoch
-     traceI tr meta "epoch"       (unEpochNo epoch)
+  :: forall blk. ()
+  => HasHeader (Header blk)
+  => BlockSupportsMetrics blk
+  => BlockConfig blk
+  -> Trace IO Text
+  -> IO (Tracer IO (ChainDB.TraceEvent blk))
+traceChainMetrics blockConfig tr = do
+  tSwitchingBlocksLost <- STM.newTVarIO (0 :: Int64)
+  return . Tracer $ \ev ->
+    fromMaybe (pure ()) $
+      doTrace tSwitchingBlocksLost <$> chainTipInformation ev
+  where
+    chainTipInformation :: ChainDB.TraceEvent blk -> Maybe ChainInformation
+    chainTipInformation = \case
+      ChainDB.TraceAddBlockEvent ev -> case ev of
+        ChainDB.SwitchedToAFork _warnings newTipInfo oldChain newChain ->
+          let blocksLost = (countMyBlocks blockConfig oldChain) - (countMyBlocks blockConfig newChain) in
+          Just $ chainInformation newTipInfo newChain blocksLost
+        ChainDB.AddedToCurrentChain _warnings newTipInfo _ chain ->
+          Just $ chainInformation newTipInfo chain 0
```

Comment:
I think you want to do the same thing here: count it as "-1 blocks lost" if the new block is one we forged.

More specifically: I'm not certain about this next claim, but I'd be surprised if the new chain had more than one additional block. So you should be able to look at just the latest header in `chain`.

Alternatively, you could so the same counting within each chain and then the subtraction as in the `SwitchedToAFork` case.

https://github.com/input-output-hk/ouroboros-network/blob/1efb25d657ca24da00fdcefa17fb6c1eb6909b49/ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/Types.hs#L592-L599

Reply:
If you don't want to consider extensions, then I think you're doing what Duncan had proposed. But he had also proposed bounding the delta at 0, so it only ever increases, IIRC. (Maybe you're chooisng some middle ground that I don't recall from the Slack thread?)

Reply:
Thanks for the suggestion.  I make the code the same as the `SwitchedToAFork` case.  I think that should work.